### PR TITLE
BIP-44 implementation

### PR DESCRIPTION
**Pull request by AmiMobilechain**
1. implement BIP-32 keypair
2. implement BIP-44 wallet on top of BIP-39 seed

### CODE REVIEW

**Comment by steveswing** [comment]
+1. This pull request will be useful.

**Comment by kwolfy** [comment]
It works for me correct. I got capability with https://github.com/D-Technologies/EthereumKit , it uses bip44 too. 
Thank you for PR

**Comment by kwolfy** [comment]
@amiperlmutter I think boolean is really good idea.

**Comment by amiperlmutter** [comment]
@kwolfy are you sure it is not /m/44**'**/0**'**/0/external ?

File: crypto/src/test/java/org/web3j/crypto/Bip32Test.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    static byte[] serializePublic(Bip32ECKeyPair pair) {
+        return serialize(pair, 0x0488B21E, true);
+    }
+
+    static byte[] serializePrivate(Bip32ECKeyPair pair) {
+        return serialize(pair, 0x0488ADE4, false);
+    }
+
+    private byte[] hexToByte(String encoded) {
```
Comment (conor10) [inline crypto/src/test/java/org/web3j/crypto/Bip32Test.java]: This function is already defined - https://github.com/web3j/web3j/blob/master/utils/src/main/java/org/web3j/utils/Numeric.java#L174

Comment (steveswing) [reply crypto/src/test/java/org/web3j/crypto/Bip32Test.java]: Should Bip32Test.java be named Bip32ECKeyPairTest.java?

Comment (AmiMobilechain) [reply crypto/src/test/java/org/web3j/crypto/Bip32Test.java]: fixed, let me know if I should also rename the file

File: crypto/src/main/java/org/web3j/crypto/Sign.java [hunk 2]
```diff
@@ -228,6 +232,17 @@ private static ECPoint publicPointFromPrivate(BigInteger privKey) {
...
     }
 
+    /**
+     * Returns public key point from the given curve.
+     *
+     * @param curve the curve
+     * @param bits representing the point on the curve
+     * @return BigInteger encoded public key
+     */
+    public static BigInteger publicFromPoint(ECCurve curve, byte[] bits) {
```
[the remark this answers is not in the document]
Comment (AmiMobilechain) [reply crypto/src/main/java/org/web3j/crypto/Sign.java]: curve removed

**Comment by AmiMobilechain** [comment]
@conor10 how does the process of porting to Android work? do you want me to port this?

**Comment by iikirilov** [comment]
@AmiMobilechain Porting to android is being discussed in #769

**Pull request merged**